test/cmake/compress-and-verify.cmake

```diff
@@ -101,24 +113,47 @@ macro(diff src1 src2)
     endif()
 endmacro()
 
+
+macro(exec_streams cmd1 src1 dst1)
+    execute_process(COMMAND ${CMAKE_COMMAND}
+        "-DCOMMAND=${cmd1}"
+        -DINPUT=${src1}
+        -DOUTPUT=${dst1}
+        "-DSUCCESS_EXIT=${SUCCESS_EXIT}"
+        -P ${CMAKE_CURRENT_LIST_DIR}/run-and-redirect.cmake
+        RESULT_VARIABLE CMD_RESULT
+        COMMAND_ECHO STDOUT)
+endmacro()
+
+macro(exec_files cmd1 src1)
+    execute_process(COMMAND
+        ${cmd1} ${src1}
+        RESULT_VARIABLE CMD_RESULT
+        COMMAND_ECHO STDOUT)
+endmacro()
+
 # Compress input file
 if(NOT EXISTS ${INPUT})
     message(FATAL_ERROR "Cannot find compress input: ${INPUT}")
 endif()
 
 set(COMPRESS_COMMAND ${COMPRESS_TARGET} ${COMPRESS_ARGS})
 
+set(INPUT_FILE ${OUTPUT_BASE})
+
+# Make CMake copy and rename file in one operation
+configure_file(${INPUT} ${INPUT_FILE} COPYONLY)
```

Comment:
Why does it need to copy file?

Reply:
Otherwise it would just compress the file in the source path to a file with the input name and only appends '.gz'.
So we need to first copy the file to the current tests temp folder and give it the expected name.